src/controller/datacontroller.js

```diff
@@ -46,11 +46,12 @@ export default class DataController {
 	/**
 	 * Creates a data controller instance.
 	 *
+	 * @param {module:engine/view/stylesmap~StylesProcessor} stylesProcessor Styles processor.
 	 * @param {module:engine/model/model~Model} model Data model.
 	 * @param {module:engine/dataprocessor/dataprocessor~DataProcessor} [dataProcessor] Data processor that should be used
 	 * by the controller.
 	 */
-	constructor( model, dataProcessor ) {
+	constructor( stylesProcessor, model, dataProcessor ) {
```

Comment:
Why is this the first param? It should be the last one. It is the last one in the `EditingController` already.

Reply:
Because `dataProcessor` is not required. Following the docs:

```
@param {module:engine/dataprocessor/dataprocessor~DataProcessor} [dataProcessor]
```

Reply:
There is also a test that says the same.

Reply:
Still, the `model` is required. So if anything, this should be the 2nd arg.